netboot/main.go

```diff
@@ -165,15 +172,50 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	}
 	debug("DHCP: boot file URL is %s", bootfile)
 	// check for supported schemes
-	if !strings.HasPrefix(bootfile, "http://") {
-		return fmt.Errorf("DHCP: can only handle http scheme")
+	matched, err := regexp.Match("^(http|https)://", []byte(bootfile))
+	if !matched || err != nil {
+		return fmt.Errorf("DHCP: can only handle http/https scheme")
+	}
+
+	if strings.HasPrefix(bootfile, "https://") {
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			debug("certs: rootCAs == nil")
+			rootCAs = x509.NewCertPool()
+		}
+		base64Certs, err := ioutil.ReadFile(*caCertFile)
+		decLen := base64.StdEncoding.DecodedLen(len(base64Certs))
+		certs := make([]byte, decLen)
+		n, err := base64.StdEncoding.Decode(certs, base64Certs)
+		if err != nil {
+			return fmt.Errorf("Failed to append %s to RootCAs: %v", *caCertFile, err)
+		}
+		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+			debug("No VPD certs appended, using system certs only")
+		} else {
+			debug("%d bytes of VPD certs appended", n)
+		}
+		config := &tls.Config{
+			InsecureSkipVerify: *skipCertVerify,
+			RootCAs:            rootCAs,
+		}
+		tr := &http.Transport{TLSClientConfig: config}
```

Comment:
Do we want to setup client-specific security requirements here?

Reply:
What do you mean ? Are there other client-specific requirements, or do you think they should be set elsewhere?